onnxruntime/core/graph/runtime_optimization_record_container.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "core/common/common.h"
+#include "core/graph/runtime_optimization_record.h"
+
+#if !defined(ORT_MINIMAL_BUILD)
+#define ORT_ENABLE_ADDING_RUNTIME_OPTIMIZATION_RECORDS
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
```

Comment:
Should the rest of this file be excluded if runtime type optimization is disabled (either via cmake or ifdefs)? Not sure if it's able to be thrown away by the linker so there's no binary size impact in a minimal build if that is the case.